OpenRA.Game/GlobalChat.cs

```diff
@@ -0,0 +1,358 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Meebey.SmartIrc4net;
+using OpenRA;
+using OpenRA.Primitives;
+
+namespace OpenRA.Chat
+{
+	public enum ChatConnectionStatus { Disconnected, Connecting, Connected, Disconnecting, Joined, Error }
+	public enum ChatMessageType { Message, Notification }
+
+	public sealed class ChatUser
+	{
+		public readonly string Name;
+		public bool IsOp;
+		public bool IsVoiced;
+
+		public ChatUser(string name, bool isOp, bool isVoice)
+		{
+			Name = name;
+			IsOp = isOp;
+			IsVoiced = isVoice;
+		}
+	}
+
+	public sealed class ChatMessage
+	{
+		static long nextUID;
+
+		public readonly DateTime Time;
+		public readonly ChatMessageType Type;
+		public readonly string Nick;
+		public readonly string Message;
+		public readonly string UID;
+
+		public ChatMessage(DateTime time, ChatMessageType type, string nick, string message)
+		{
+			Time = time;
+			Type = type;
+			Nick = nick;
+			Message = message;
+
+			UID = (++nextUID).ToString();
+		}
+
+		public override string ToString()
+		{
+			var time = Time.ToString(Game.Settings.Chat.TimestampFormat);
+			if (Type == ChatMessageType.Notification)
+				return "{0} {1}".F(time, Message);
+
+			return "{0} {1}: {2}".F(time, Nick, Message);
+		}
+	}
+
+	public class GlobalChat : IDisposable
+	{
+		volatile IrcClient client;
```

Comment:
This was carried over from the earlier code, where I believe you asked for it yourself :P.
But you're right - this is only ever accessed from the worker thread now.